### PR TITLE
DOC: test organization

### PR DESCRIPTION
**Pull request by jbrockmendel**
Retry of #37637, reverted by #37756

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@jorisvandenbossche thoughts on the rst?

File: doc/source/development/test_writing.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
+.. _test_organization:
+
+Test organization
+=================
+Ideally, there should be one, and only one, obvious place for a test to reside.
+Until we reach that ideal, these are some rules of thumb for where a test should
+be located.
+
+1. Does your test depend only on code in ``pd._libs.tslibs``?
+This test likely belongs in one of:
```
Comment (jorisvandenbossche) [inline doc/source/development/test_writing.rst:10]: This line needs to be aligned with the "Does .." from the line above. See eg https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks for rst syntax for lists (and nested lists)

File: doc/source/development/test_writing.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    - tests.tslibs
```
Comment (jorisvandenbossche) [inline doc/source/development/test_writing.rst:12]: This also needs to be aligned with the "Does .." from the first line of this bullet point (the doc link I gave above also has an example of nested lists)

File: doc/source/development/test_writing.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        .. note::
```
Comment (jorisvandenbossche) [inline doc/source/development/test_writing.rst:14]: This note is part of the "tests.tslibs" text block, so the start of this (the ..) needs to align with "tests.tslibs"

File: doc/source/development/test_writing.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+             expected = pd.Series([2, 3])
+             tm.assert_series_equal(result, expected)
+
+             result = ser.loc[[3, 4]]
+             tm.assert_series_equal(result, expected)
+
+    In cases like this, the test location should be based on the *underlying* method being tested.  Or in the case of a test for a bugfix, the location of the actual bug.  So in this example, we know that ``Series.__getitem__`` calls ``Series.loc.__getitem__``, so this is *really* a test for ``loc.__getitem__``.  So this test belongs in ``tests.indexing.test_loc``
+
+6. Is your test for a DataFrame or Series method?
+    A) Is the method a plotting method?
```
Comment (jorisvandenbossche) [inline doc/source/development/test_writing.rst:113]: When starting a sub-item (the A) list item), there needs to be a blank line preceding it to make the separation with the previous (parent) item clear

File: doc/source/development/test_writing.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+             ser = pd.Series(range(5))
+
+             result = ser[[3, 4]]
+             expected = pd.Series([2, 3])
+             tm.assert_series_equal(result, expected)
+
+             result = ser.loc[[3, 4]]
+             tm.assert_series_equal(result, expected)
+
+    In cases like this, the test location should be based on the *underlying* method being tested.  Or in the case of a test for a bugfix, the location of the actual bug.  So in this example, we know that ``Series.__getitem__`` calls ``Series.loc.__getitem__``, so this is *really* a test for ``loc.__getitem__``.  So this test belongs in ``tests.indexing.test_loc``
```
Comment (jorisvandenbossche) [inline doc/source/development/test_writing.rst:110]: Can you hard-wrap those lines?

**Review by jorisvandenbossche — COMMENTED** [review]
Didn't look yet at the actual content, but a few comments on the rest markup (I commented mostly on the first bullet point, but it needs to be fixed for the full file, to be clear)

**Comment by jbrockmendel** [comment]
@jorisvandenbossche thanks for the notes, implementing them now

**Comment by jreback** [comment]
thanks @jbrockmendel 

merging this, will likely be updating as we go so if more comments can handle then.

**Pull request merged**